favorites.py

```diff
@@ -103,12 +103,24 @@ class FavFileMgr(object):
     """Handle file actions."""
 
     @classmethod
+    def fav_file(cls, filename, data=False, read=False, write=False):
+        """Read or write currently handled favorite list."""
+
+        if read:
+            with open(filename) as file:
+                # Allow C style comments and be forgiving of trailing commas
+                content = sanitize_json(file.read(), True)
+            return json.loads(content)
+
+        elif write:
+            with open(filename, "w") as file:
+                json.dump(data, file, sort_keys=True, indent=4, separators=(',', ': '))
+

```

Comment:
I made a function for basic access to json file, that returns the contents or saves it. There were more places where the file had to be read and updated (check_aliases, and remember toggle state).